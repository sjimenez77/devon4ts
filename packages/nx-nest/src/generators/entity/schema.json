{
  "$schema": "https://json-schema.org/schema",
  "$id": "Entity",
  "title": "",
  "type": "object",
  "properties": {
    "name": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "What name would you like to use?"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 1
      },
      "x-prompt": "In which project would you like to add the entity?"
    },
    "module": {
      "type": "string",
      "description": "Module where the entity will be generated"
    }
  },
  "required": ["name", "projectName"]
}
