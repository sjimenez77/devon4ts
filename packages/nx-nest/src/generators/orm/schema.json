{
  "$schema": "http://json-schema.org/schema",
  "$id": "InitTypeormSchematic",
  "title": "Init Typeorm Schema",
  "type": "object",
  "properties": {
    "orm": {
      "type": "string",
      "enum": ["typeorm"],
      "default": "typeorm"
    },
    "db": {
      "type": "string",
      "enum": ["postgres", "cockroachdb", "mariadb", "mysql", "sqlite", "oracle", "mssql", "mongodb"],
      "default": "sqlite"
    },
    "projectName": {
      "type": "string",
      "description": "",
      "$default": {
        "$source": "argv",
        "index": 0
      },
      "x-prompt": "In which project would you like to add typeorm configuration?"
    }
  },
  "required": ["projectName"]
}
