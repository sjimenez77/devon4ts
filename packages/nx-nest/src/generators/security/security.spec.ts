import { Tree, readProjectConfiguration } from '@nx/devkit';
import { createTreeWithEmptyWorkspace } from '@nx/devkit/testing';
import applicationGenerator from '../application/application';
import { SecurityGeneratorSchema } from './schema';
import { securityGenerator } from './security';

describe('Security Generator', () => {
  let tree: Tree;
  const options: SecurityGeneratorSchema = { projectName: 'test' };

  beforeAll(async () => {
    tree = createTreeWithEmptyWorkspace();
    await applicationGenerator(tree, {
      name: options.projectName,
      projectNameAndRootFormat: 'as-provided',
      directory: 'apps/test',
    });
    await securityGenerator(tree, options);
  }, 60000);

  it('should run successfully', async () => {
    const config = readProjectConfiguration(tree, 'test');
    expect(config).toBeDefined();
  });

  it('should add dependencies to package.json', async () => {
    const fileContent = tree.read('package.json')?.toString('utf-8');
    expect(fileContent).toMatch(/"dependencies": {(.|\n)*"helmet":/g);
  });

  it('should add CORS and helmet to main.ts', async () => {
    const fileContent = tree.read(`./packages/nx-nest/apps/${options.projectName}/src/main.ts`)?.toString('utf-8');
    expect(fileContent).toMatchSnapshot();
  });
});
