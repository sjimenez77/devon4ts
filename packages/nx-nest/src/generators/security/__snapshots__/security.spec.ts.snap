// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Security Generator should add CORS and helmet to main.ts 1`] = `
"/**
 * This is not a production server yet!
 * This is only a minimal backend to get started.
 */

import { Logger, ValidationPipe, VersioningType } from '@nestjs/common';
import { NestFactory } from '@nestjs/core';

import { AppModule } from './app/app.module';
import { WinstonLogger } from '@proj/shared/logger';
import helmet from 'helmet';

async function bootstrap(): Promise<void> {
  const app = await NestFactory.create(AppModule, { bufferLogs: true });

  const logger = await app.resolve(WinstonLogger);
  app.useLogger(logger);

  const globalPrefix = 'api';
  app.setGlobalPrefix(globalPrefix);
  const port = process.env.PORT ?? 3000;
  app.enableCors({
    origin: '*',
    credentials: true,
    exposedHeaders: 'Authorization',
    allowedHeaders: 'authorization, content-type',
  });
  app.use(helmet());
  app.useGlobalPipes(
    new ValidationPipe({
      transform: true,
      transformOptions: {
        excludeExtraneousValues: true,
      },
    }),
  );
  app.enableVersioning({
    type: VersioningType.URI,
    defaultVersion: '1',
  });
  await app.listen(port);
  Logger.log(\`ðŸš€ Application is running on: http://localhost:\${port}/\${globalPrefix}\`);
}

bootstrap();
"
`;
